src/main/java/uk/gov/hmcts/darts/retention/helper/RetentionDateHelper.java

```diff
@@ -45,25 +46,38 @@ public RetentionPolicyTypeEntity getRetentionPolicy(RetentionPolicyEnum policy)
         return manualPolicyEntityOpt.get();
     }
 
-    private LocalDate applyPolicyString(LocalDate dateToAppend, String policyString) {
-        String[] policyArray = StringUtils.splitByCharacterType(StringUtils.trimToEmpty(policyString));
+    public LocalDate applyPolicyString(LocalDate dateToAppend, String policyString) {
+        if (StringUtils.isBlank(policyString) || !policyFormat.matcher(policyString).matches()) {
+            throw new DartsApiException(
+                    RetentionApiError.INTERNAL_SERVER_ERROR,
+                    MessageFormat.format("PolicyString ''{0}'', is not in the required format.", policyString)
+            );
+        }
 
-        int years = getValueFromPolicyArray(policyArray, "Y");
-        int months = getValueFromPolicyArray(policyArray, "M");
-        int days = getValueFromPolicyArray(policyArray, "D");
+        String[] policyArray = StringUtils.splitByCharacterType(StringUtils.trimToEmpty(policyString));
 
-        LocalDate newDate = dateToAppend.plusYears(years);
-        newDate = newDate.plusMonths(months);
-        newDate = newDate.plusDays(days);
+        LocalDate newDate = dateToAppend.plusYears(NumberUtils.toInt(policyArray[0]));
```

Comment:
Avoid magic numbers in favour of more descriptive constants

Reply:
done